Fix function tests flaps

azat opened the pull request:
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Changelog:
- 01548_query_log_query_execution_ms (retries)
- 01675_distributed_bytes_to_delay_insert (retries)

azat committed:
Fix 01548_query_log_query_execution_ms flap (by increasing the time delta)

From server logs in [1]:

    2021.04.22 10:09:12.216874 [ 19931 ] {cad834d6-e01d-4695-ad19-0bc6cdff65c3} <Debug> executeQuery: (from [::1]:37534, using production parser) (comment: /usr/share/clickhouse-test/queries/0_stateless/01548_query_log_query_execution_ms.sql) select '01548_query_log_query_execution_ms', sleep(0.4) format Null;
    ...
    2021.04.22 10:09:13.495662 [ 19931 ] {cad834d6-e01d-4695-ad19-0bc6cdff65c3} <Information> executeQuery: Read 1 rows, 1.00 B in 1.27832443 sec., 0 rows/sec., 0.78 B/sec.

  [1]: https://clickhouse-test-reports.s3.yandex.net/23469/2134a4f4c16da093194486ead133afae5cb4e827/functional_stateless_tests_(thread).html#fail1

akuzm commented on tests/queries/0_stateless/01548_query_log_query_execution_ms.sql:
```diff
@@ -12,7 +12,7 @@ from system.query_log
     query like '%01548_query_log_query_execution_ms%'
     and current_database = currentDatabase()
-    and query_duration_ms between 100 and 800
+    and query_duration_ms between 400 and 3000
```
Why do you need an upper limit on the query duration? Maybe lower limit of 0.4 seconds is enough?

azat replied:
I decided to leave the upper limit to ensure that 0.4 will not be 4 seconds (by some strange reason), but I'm fine with removing it altogether

azat committed:
Convert 01548_query_log_query_execution_ms to .sh with endless retries

alexey-milovidov commented:
Ok. Now `01675_distributed_bytes_to_delay_insert` is flaky.

azat committed:
Add retries into 01675_distributed_bytes_to_delay_insert (and tune delays)

azat commented:
>Ok. Now 01675_distributed_bytes_to_delay_insert is flaky.

Yes, that's why I've marked as draft, now 01675_distributed_bytes_to_delay_insert also fixed

azat committed:
Mark 01675_distributed_bytes_to_delay_insert as long

https://clickhouse-test-reports.s3.yandex.net/23517/416bacc341a815b1b2d28ae24a31e13982ff5a0f/functional_stateless_tests_flaky_check_(address)/test_run.txt.out.log

azat committed:
Improve retry logic in 01675_distributed_bytes_to_delay_insert_long

https://clickhouse-test-reports.s3.yandex.net/23517/414d8d0634ea5e3f05d30f53df8531da54a7ff50/functional_stateless_tests_(ubsan).html#fail1

azat commented:
>[AST fuzzer (debug) — Fuzzer failed (1). See the logs.](https://clickhouse-test-reports.s3.yandex.net/23517/f1187aeb69109c88f0be978b8083080c7a843820/fuzzer_debug/report.html#fail1)

#23595

alexey-milovidov commented:
@mergifyio update

mergify[bot] committed:
Merge branch 'master' into flaky-tests-v21.6

alexey-milovidov commented:
@Mergifyio update

mergify[bot] committed:
Merge branch 'master' into flaky-tests-v21.6

alexey-milovidov commented:
```
2021-04-25 03:04:54
Unable to continue: Traceback (most recent call last):
  File "sandbox/bin/executor.py", line 1033, in _execute
    self.task.on_execute()
  File "sandbox/projects/clickhouse/BaseOnCommitTask/base.py", line 304, in on_execute
    state, description, url = self.process(self.commit, repo, self.pull_request)
  File "sandbox/projects/clickhouse/ClickhouseBuilder/__init__.py", line 469, in process
    ccache_path = self._get_cache_path(commit, pull_request)
  File "sandbox/projects/clickhouse/ClickhouseBuilder/__init__.py", line 264, in _get_cache_path
    cache_data = sdk2.ResourceData(result)
  File "sandbox/sdk2/resource.py", line 203, in __init__
    super(ResourceData, self).__init__(resource, fastbone)
  File "sandbox/sdk2/internal/resource.py", line 751, in __init__
    self.__path = sdk2.Path(sdk2.Task.current.agentr.resource_sync(self.__resource.id, **kws))
  File "sandbox/agentr/client/__init__.py", line 310, in resource_sync
    return self.local_path(self("resource_sync", rid, fastbone=fastbone, restore=restore))
  File "sandbox/agentr/client/__init__.py", line 278, in __call__
    return c.wait()
  File "sandbox/common/joint/client.py", line 358, in wait
    data = self._get()
  File "sandbox/common/joint/client.py", line 405, in _get
    return self._get()
  File "sandbox/common/joint/client.py", line 425, in _get
    cls=cls_name, tb=data[5], propagate=True
ResourceNotAvailable: {da073518:9}: sandbox.agentr.errors.ResourceNotAvailable("Resource #2124877580 is not available at all sources [u'sandbox-yabs98'].",)
```

alexey-milovidov commented:
A bug in testing infrastructure: resource with ccache is "found" but cannot download:

> Found ccache resource id 2124877580 from PR 23517 @ f1187aeb69109c88f0be978b8083080c7a843820 by attrs filter {'bundled': u'bundled', 'with_coverage': False, 'splitted': u'unsplitted', 'tidy': u'disable', 'package_type': u'binary', 'sanitizer': 'none', 'build_type': 'relwithdebuginfo', 'pr_number': 23517, 'compiler': u'clang-11'}

alexey-milovidov commented:
SQLancer: https://github.com/sqlancer/sqlancer/pull/340
@mrigger please help merging.

alexey-milovidov commented:
@azat Now this test has failed: 01641_memory_tracking_insert_optimize_long
https://clickhouse-test-reports.s3.yandex.net/0/1c472b014d814145023e41cbf437dac5a9dcb494/functional_stateless_tests_(release,_databaseordinary).html#fail1

alexey-milovidov commented:
#23617

mrigger commented:
> SQLancer: [sqlancer/sqlancer#340](https://github.com/sqlancer/sqlancer/pull/340)
> @mrigger please help merging.

Sorry for the delay, the PR is merged now.
